From 118b8240476469d3fb9fff1c035646fe4cb03e0b Mon Sep 17 00:00:00 2001
From: firmwarecostum <145313986+firmwarecostum@users.noreply.github.com>
Date: Mon, 13 Oct 2025 19:11:48 +0800
Subject: [PATCH] Update Make-lang.in

---
 gcc/go/Make-lang.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/go/Make-lang.in b/gcc/go/Make-lang.in
index a1f2977965e82..ca7a4d9e8f4c2 100644
--- a/gcc/go/Make-lang.in
+++ b/gcc/go/Make-lang.in
@@ -83,7 +83,7 @@ go_OBJS = $(GO_OBJS) go/gospec.o
 go1$(exeext): $(GO_OBJS) attribs.o $(BACKEND) $(LIBDEPS) $(go.prev)
 	@$(call LINK_PROGRESS,$(INDEX.go),start)
 	+$(LLINKER) $(ALL_LINKERFLAGS) $(LDFLAGS) -o $@ \
-	      $(GO_OBJS) attribs.o $(BACKEND) $(LIBS) $(BACKENDLIBS)
+	      $(GO_OBJS) attribs.o $(BACKEND) $(LIBS) $(BACKENDLIBS) -lucontext -lucontext_posix
 	@$(call LINK_PROGRESS,$(INDEX.go),end)
 
 # Documentation.
