From be8433e826a95135601ec920e3b1a6a39cb09dbc Mon Sep 17 00:00:00 2001
From: Predators <31931530+predators46@users.noreply.github.com>
Date: Tue, 30 Sep 2025 15:26:33 +0800
Subject: [PATCH] Update go-signal.c

---
 libgo/runtime/go-signal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgo/runtime/go-signal.c b/libgo/runtime/go-signal.c
index aa1b6305ad09d..20d01a683089b 100644
--- a/libgo/runtime/go-signal.c
+++ b/libgo/runtime/go-signal.c
@@ -8,7 +8,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <sys/time.h>
-#include <ucontext.h>
+#include <signal.h>
 
 #include "runtime.h"
 
From bf5cc82524336994bbdcd71e27817f4bf961e9fd Mon Sep 17 00:00:00 2001
From: Predators <31931530+predators46@users.noreply.github.com>
Date: Tue, 30 Sep 2025 15:29:51 +0800
Subject: [PATCH] Update runtime.h

---
 libgo/runtime/runtime.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libgo/runtime/runtime.h b/libgo/runtime/runtime.h
index 699770d53add4..4ca2ee6e5b766 100644
--- a/libgo/runtime/runtime.h
+++ b/libgo/runtime/runtime.h
@@ -17,7 +17,6 @@
 #include <unistd.h>
 #include <pthread.h>
 #include <semaphore.h>
-#include <ucontext.h>
 
 #ifdef HAVE_SYS_MMAN_H
 #include <sys/mman.h>

From ebde00f328334a871499478ce7ca6ac229427144 Mon Sep 17 00:00:00 2001
From: Predators <31931530+predators46@users.noreply.github.com>
Date: Tue, 30 Sep 2025 15:33:32 +0800
Subject: [PATCH] Update sysinfo.c

---
 libgo/sysinfo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgo/sysinfo.c b/libgo/sysinfo.c
index 180f5c31d7427..ff111006f3a83 100644
--- a/libgo/sysinfo.c
+++ b/libgo/sysinfo.c
@@ -16,7 +16,7 @@
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <ucontext.h>
+#include <signal.h>
 #include <netinet/in.h>
 /* <netinet/tcp.h> needs u_char/u_short, but <sys/bsd_types> is only
    included by <netinet/in.h> if _SGIAPI (i.e. _SGI_SOURCE

From c5cf8f817f1f6bb7b8dc45b2e28cb3702f6d2691 Mon Sep 17 00:00:00 2001
From: Predators <31931530+predators46@users.noreply.github.com>
Date: Tue, 30 Sep 2025 15:44:37 +0800
Subject: [PATCH] Update tsan_shared.go

---
 libgo/misc/cgo/testsanitizers/testdata/tsan_shared.go | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libgo/misc/cgo/testsanitizers/testdata/tsan_shared.go b/libgo/misc/cgo/testsanitizers/testdata/tsan_shared.go
index 55ff67ecbafc4..364d089d6a032 100644
--- a/libgo/misc/cgo/testsanitizers/testdata/tsan_shared.go
+++ b/libgo/misc/cgo/testsanitizers/testdata/tsan_shared.go
@@ -17,7 +17,6 @@ package main
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <ucontext.h>
 
 void check_params(int signo, siginfo_t *info, void *context) {
 	ucontext_t* uc = (ucontext_t*)(context);

From b5bbdf097620d19c849271ba544ed955aab6d8cd Mon Sep 17 00:00:00 2001
From: Predators <31931530+predators46@users.noreply.github.com>
Date: Tue, 30 Sep 2025 15:57:29 +0800
Subject: [PATCH] Update configure.ac

---
 libgo/configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgo/configure.ac b/libgo/configure.ac
index a59aa091d1dce..f1dbb495ac6fa 100644
--- a/libgo/configure.ac
+++ b/libgo/configure.ac
@@ -720,7 +720,7 @@ AC_RUN_IFELSE(
   [AC_LANG_SOURCE([
 #include <pthread.h>
 #include <stdlib.h>
-#include <ucontext.h>
+#include <signal.h>
 #include <unistd.h>
 
 __thread int tls;

From 00878f9a57857bf401c1f2a80816e83a25c2ff32 Mon Sep 17 00:00:00 2001
From: esaaprillia <esaapriliasalsabila@gmail.com>
Date: Tue, 30 Sep 2025 16:02:23 +0800
Subject: [PATCH] Update configure

---
 libgo/configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgo/configure b/libgo/configure
index a607dbf..0eea711 100644
--- a/libgo/configure
+++ b/libgo/configure
@@ -15868,7 +15868,7 @@ else
 
 #include <pthread.h>
 #include <stdlib.h>
-#include <ucontext.h>
+#include <signal.h>
 #include <unistd.h>
 
 __thread int tls;

--- a/libgo/runtime/go-callers.c.old	2023-01-01 17:56:27.302982459 -0600
+++ b/libgo/runtime/go-callers.c	2023-01-01 17:58:00.612903483 -0600
@@ -165,26 +165,11 @@
   if (function != NULL)
     {
-      if (__builtin_strcmp (function, "makecontext") == 0)
+      if (__builtin_strcmp (function, "makecontext") == 0
+          || __builtin_strcmp (function, "runtime_mstart") == 0
+          || __builtin_strcmp (function, "runtime.kickoff") == 0
+          || __builtin_strcmp (function, "runtime.main") == 0)
 	return 1;
-      if (filename != NULL)
-	{
-	  const char *p;
-
-	  p = strrchr (filename, '/');
-	  if (p == NULL)
-	    p = filename;
-	  if (__builtin_strcmp (p, "/proc.c") == 0)
-	    {
-	      if (__builtin_strcmp (function, "runtime_mstart") == 0)
-		return 1;
-	    }
-	  else if (__builtin_strcmp (p, "/proc.go") == 0)
-	    {
-	      if (__builtin_strcmp (function, "runtime.kickoff") == 0
-		  || __builtin_strcmp (function, "runtime.main") == 0)
-		return 1;
-	    }
-	}
     }
 
   return arg->index >= arg->max;

