From b71fa3682f12482753f432ef6c5c3c391c699259 Mon Sep 17 00:00:00 2001
From: esaaprillia <esaapriliasalsabila@gmail.com>
Date: Sun, 28 Sep 2025 21:39:59 +0800
Subject: [PATCH] Update configure

---
 libgo/configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgo/configure b/libgo/configure
index a607dbf..d9b522c 100644
--- a/libgo/configure
+++ b/libgo/configure
@@ -14847,7 +14847,7 @@ for ac_lib in '' ucontext; do
     ac_res="none required"
   else
     ac_res=-l$ac_lib
-    LIBS="-l$ac_lib  $ac_func_search_save_LIBS"
+    LIBS="-lucontext  $ac_func_search_save_LIBS"
   fi
   if ac_fn_c_try_link "$LINENO"; then :
   ac_cv_search_makecontext=$ac_res
