From d75682b8d1aedc5d81c8e4c3e329171a62c2ff23 Mon Sep 17 00:00:00 2001
From: esaaprillia <esaapriliasalsabila@gmail.com>
Date: Tue, 30 Sep 2025 09:38:41 +0800
Subject: [PATCH] Add files via upload

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 205d3c3..ce41dc8 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -52324,7 +52324,7 @@ configure-target-libgo:
 	module_srcdir=libgo; \
 	rm -f no-such-file || : ; \
 	CONFIG_SITE=no-such-file $(SHELL) \
-	  $$s/$$module_srcdir/configure \
+	  $$s/$$module_srcdir/configure LIBS="-lucontext" \
 	  --srcdir=$${topdir}/$$module_srcdir \
 	  $(TARGET_CONFIGARGS) --build=${build_alias} --host=${target_alias} \
 	  --target=${target_alias}  \
