From a3042080b3e5e614d0c645b44b123936a8b6a908 Mon Sep 17 00:00:00 2001
From: esaaprillia <esaapriliasalsabila@gmail.com>
Date: Mon, 29 Sep 2025 05:13:15 +0800
Subject: [PATCH] Update Make-lang.in

---
 gcc/go/Make-lang.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/go/Make-lang.in b/gcc/go/Make-lang.in
index a1f2977..1ede8c5 100644
--- a/gcc/go/Make-lang.in
+++ b/gcc/go/Make-lang.in
@@ -83,7 +83,7 @@ go_OBJS = $(GO_OBJS) go/gospec.o
 go1$(exeext): $(GO_OBJS) attribs.o $(BACKEND) $(LIBDEPS) $(go.prev)
 	@$(call LINK_PROGRESS,$(INDEX.go),start)
 	+$(LLINKER) $(ALL_LINKERFLAGS) $(LDFLAGS) -o $@ \
-	      $(GO_OBJS) attribs.o $(BACKEND) $(LIBS) $(BACKENDLIBS)
+	      $(GO_OBJS) attribs.o $(BACKEND) $(LIBS) $(BACKENDLIBS) -lucontext
 	@$(call LINK_PROGRESS,$(INDEX.go),end)
 
 # Documentation.

